src/insights_client/metrics.py

```diff
@@ -9,18 +9,27 @@
 
 
 def _proxy_settings(rhsm_config):
+    conf = {"hostname": None,
+            "port": None,
+            "user": None,
+            "password": None}
+
     try:
-        hostname = rhsm_config.get("server", "proxy_hostname").strip()
-        if not hostname:
-            return None
-        port = rhsm_config.get("server", "proxy_port").strip()
-        user = rhsm_config.get("server", "proxy_user").strip()
-        password = rhsm_config.get("server", "proxy_password").strip()
-    except (configparser.NoSectionError, configparser.NoOptionError):
+        for key in conf:
+            try:
+                conf[key] = rhsm_config.get("server", "proxy_" + key).strip()
+                if key == "hostname" and conf[key] is None:
+                    # hostname is the only required value. return None if empty
+                    return None
+            except configparser.NoOptionError:
+                if key == "hostname":
+                    return None
+    except configparser.NoSectionError:
```

Comment:
This is not tested at all. All test cases have the _server_ section present.

Reply:
Getting there. I just wanted your thoughts on the syntax before I went all-in on it.